### PR TITLE
Tidy up some Guice APIs

### PR DESCRIPTION
**Pull request by swankjesse**
Be more consistent with multibindings, and avoid needing to deference
the binder in the common case that we're in a KAbstractModule.

### CODE REVIEW

File: misk/src/main/kotlin/misk/inject/KAbstractModule.kt [hunk 1]
```diff
@@ -34,13 +38,33 @@ abstract class KAbstractModule : AbstractModule() {
...
-  protected inline fun <reified T : Any> AnnotatedBindingBuilder<in T>.to(): ScopedBindingBuilder {
+  protected inline fun <reified T : Any> LinkedBindingBuilder<in T>.to(): ScopedBindingBuilder {
     return to(T::class.java)
   }
 
-  protected inline fun <reified T : Any> newSetBinder(): Multibinder<T> {
-    return Multibinder.newSetBinder(binder(), T::class.java)
-  }
-}
+  protected inline fun <reified T : Any> addMultibinderBinding(
```
Comment (ryanhall07) [inline misk/src/main/kotlin/misk/inject/KAbstractModule.kt]: nit: `addMultiBinding`?

File: misk/src/main/kotlin/misk/MiskApplication.kt [hunk 2]
```diff
@@ -16,7 +16,7 @@ class MiskApplication(private val modules: List<Module>, commands: List<MiskComm
   constructor(vararg modules: Module) : this(modules.toList())
   constructor(vararg commands: MiskCommand) : this(listOf(), commands.toList())
 
-  private val commands = commands.map { it.name to it }.toMap()
+  private val commands = commands.associateBy { it.name }
```
Comment (tirsen) [inline misk/src/main/kotlin/misk/MiskApplication.kt]: Ohhh nice!

File: misk/src/main/kotlin/misk/concurrent/WrappingListeningExecutorService.kt [hunk 3]
```diff
@@ -20,7 +20,7 @@ abstract class WrappingListeningExecutorService : ForwardingListeningExecutorSer
   }
 
   override fun <T> submit(runnable: Runnable, result: T): ListenableFuture<T> {
-    return submit<T>({ runnable.run(); result })
+    return submit<T> { runnable.run(); result }
```
Comment (adrw) [inline misk/src/main/kotlin/misk/concurrent/WrappingListeningExecutorService.kt]: Kotlin-esque 👍

File: misk/src/main/kotlin/misk/inject/KAbstractModule.kt (hunk 1 above)
Comment (adrw) [reply misk/src/main/kotlin/misk/inject/KAbstractModule.kt]: @swankjesse thoughts?

Comment (swankjesse) [reply misk/src/main/kotlin/misk/inject/KAbstractModule.kt]: Yeah, that’s better

Comment (swankjesse) [reply misk/src/main/kotlin/misk/inject/KAbstractModule.kt]: Renamed to `multibind()`.

**Pull request merged**